axiom/util_strings.h

```diff
@@ -438,4 +439,36 @@ static inline int nr_toupper(int c) {
   return c;
 }
 
+/*
+ * Purpose : Checks whether a string ends with the specified pattern. Note that
+ * nr_striendswith is case-insensitive.
+ *
+ * Params  : 1. The input string to examine.
+ *           2. The input string length.
+ *           3. The pattern to look for at the end of the string.
+ *           4. The pattern length.
+ *
+ * Returns : The true if input string ends with the pattern or false otherwise.
+ */
+static inline bool nr_striendswith(const char* s,
+                                   const size_t slen,
+                                   const char* pattern,
+                                   const size_t pattern_len) {
+  const char* suffix;
+
+  if (slen < pattern_len) {
+    /* input shorter than pattern */
+    return false;
+  }
+
+  if (NULL == s) {
+    /* invalid input */
+    return false;
+  }
+
```

Comment:
What happens if `pattern` is `NULL`?

Reply:
`nr_stricmp` handles this case and returns `1` which will result in `nr_striendswith` returning `false`. I will add a test case for this.

Reply:
See 962c98c68f28cef52cd8394010b59c16d8798e02